app/controllers/guide_article_drafts_controller.ts

```diff
@@ -0,0 +1,260 @@
+      throw new ForbiddenException();
+    }
+  }
+
+  $configureRoutes(
+    controller: LazyImport<
+      Constructor<BaseController<typeof GuideArticleDraft>>
+    >,
+  ) {
+    super.$configureRoutes(controller);
+    router
+      .post("/:id/approve", async (ctx) => {
+        const module = await controller();
+        const ControllerClass = module.default;
+        const instance = new ControllerClass();
+        return (
+          instance as unknown as {
+            approve: (c: HttpContext) => Promise<unknown>;
+          }
+        ).approve(ctx as unknown as HttpContext);
+      })
+      .as("approve");
+  }
+
+  async approve({ params, auth }: HttpContext) {
+    if (!auth.isAuthenticated) {
+      await auth.authenticate();
+    }
+    // Only solvro_admin can approve drafts
+    const isSolvroAdmin = await (
+      auth.user as unknown as { hasRole?: (slug: string) => Promise<boolean> }
+    ).hasRole?.("solvro_admin");
+    if (isSolvroAdmin !== true) {
+      const { ForbiddenException } = await import(
+        "#exceptions/http_exceptions"
+      );
+      throw new ForbiddenException();
+    }
+
+    const rawId: unknown = (params as unknown as { id?: string | number }).id;
+    let draftId: number | undefined;
+    if (typeof rawId === "string") {
+      const numeric = Number(rawId);
+      draftId = Number.isNaN(numeric) ? undefined : numeric;
+    } else if (typeof rawId === "number") {
+      draftId = rawId;
+    }
+    if (draftId === undefined) {
+      const { NotFoundException } = await import("#exceptions/http_exceptions");
+      throw new NotFoundException();
+    }
+    const draft = await GuideArticleDraft.find(draftId);
+    if (draft === null) {
+      const { NotFoundException } = await import("#exceptions/http_exceptions");
+      throw new NotFoundException();
+    }
+
+    const draftData: PartialModel<typeof GuideArticle> = {
+      title: draft.title,
+      shortDesc: draft.shortDesc,
+      description: draft.description,
+    };
+    if (draft.imageKey !== null) {
+      draftData.imageKey = draft.imageKey;
+    }
+
+    let article: GuideArticle;
+    if (draft.originalArticleId !== null) {
+      const existingArticle = await GuideArticle.find(draft.originalArticleId);
+      if (existingArticle === null) {
+        const { NotFoundException } = await import(
+          "#exceptions/http_exceptions"
+        );
+        throw new NotFoundException(
+          `Original article with id ${draft.originalArticleId} not found`,
+        );
+      }
+      existingArticle.merge(draftData);
+      await existingArticle.save();
+      article = existingArticle;
+    } else {
+      if (draft.imageKey === null) {
+        const { BadRequestException } = await import(
+          "#exceptions/http_exceptions"
+        );
+        throw new BadRequestException("Cannot create article without image");
+      }
+      article = await GuideArticle.create(draftData);
+    }
+
+    await draft.delete();
```

Comment:
add error context here
also if this fails, we throw an error, the draft still exists but also the guide article gets updated...
is this the time to use a database transaction?